blocks/bricks/recurrent.py

```diff
@@ -991,8 +993,9 @@ def get_dim(self, name):
         return transition.get_dim(name)
 
     @application
-    def initial_state(self, state_name, batch_size, *args, **kwargs):
-        state_name, level = self.split_suffix(state_name)
-        transition = self.transitions[level]
-        return transition.initial_state(state_name, batch_size,
-                                        *args, **kwargs)
+    def initial_states(self, batch_size, *args, **kwargs):
+        kwargs['as_list'] = True
+        results = []
+        for transition in self.transitions:
+            results += transition.initial_states(batch_size, *args, **kwargs)
```

Comment:
Ok, I see how it works now, my bad. It would be a bit clearer to just add `as_list=True` to the function call after `batch_size`.